libnavigation-core/src/main/java/com/mapbox/navigation/core/history/MapboxHistoryRecorder.kt

```diff
@@ -4,38 +4,21 @@ import com.mapbox.base.common.logger.Logger
 import com.mapbox.base.common.logger.model.Message
 import com.mapbox.base.common.logger.model.Tag
 import com.mapbox.navigation.base.options.HistoryRecorderOptions
-import com.mapbox.navigation.base.options.NavigationOptions
 import com.mapbox.navigation.core.MapboxNavigation
 import com.mapbox.navigator.HistoryRecorderHandle
 
 /**
  * Provides a mechanism for saving and retrieving Mapbox navigation history files.
  * Retrieve an instance of this class through [MapboxNavigation.historyRecorder].
  *
- * @see [HistoryRecorderOptions] to enable the recorder.
+ * @see [HistoryRecorderOptions] to set history files directory.
  * @see [MapboxHistoryReader] to read the history files.
  */
 class MapboxHistoryRecorder internal constructor(
-    navigationOptions: NavigationOptions,
     private val logger: Logger
 ) {
-    private val historyRecorderOptions = navigationOptions.historyRecorderOptions
-    private val historyFiles = HistoryFiles(navigationOptions.applicationContext, logger)
-
     internal var historyRecorderHandle: HistoryRecorderHandle? = null
 
-    /**
-     * The file directory where the history files are stored.
-     * Use this to purge old history files, upload later, or
-     * to use them for replay.
-     *
-     * @see [HistoryRecorderOptions] to customize this location.
-     *
-     * @return absolute path to the directory with history files.
-     *     returns null when there is no file directory
-     */
-    fun fileDirectory(): String? = historyFiles.absolutePath(historyRecorderOptions)
```

Comment:
Why delete this function? I want to use it for a few reasons

1. Create a list of files that can be replayed https://github.com/mapbox/mapbox-navigation-android/pull/4601
2. Allow downstream to delete old history files
3. Provide a mechanism (not just logging) to check if the fileDirectory is indeed hooked up correctly

Reply:
it's deleted because `NavigationFiles.historyAbsolutePath` does exactly the same.

Reply:
returned param

Reply:
But this is the public interface. It's not about same functionality, it's purpose is to expose the interface in one place.

Reply:
Can see that, instead it is to expose a public `val fileDirectory`. Please make it nullable